.github/workflows/internal-ci-checks.yml

```
@@ -0,0 +1,31 @@
+name: Internal CI Checks
+# This workflows checks if the author of a PR is a member of the CodeQL team
+# and adds `ready-for-internal-ci` label to trigger the internal CI
+
+on:
+  pull_request_target:
+
+permissions:
+  pull-requests: write
```

I think we need `read:org` permissions for the team membership check.
```suggestion
  pull-requests: write
  org: read
```

---

Although that permission is not mentioned in the Actions docs: https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs